cluster/src/main/java/io/scalecube/cluster/Cluster.java

```diff
@@ -178,14 +176,8 @@ public static ICluster joinAwait(ClusterConfig config) {
         failureDetector.start();
         gossip.start();
         return membership.start();
-      });
-
-    return transform(clusterFuture, new Function<Void, ICluster>() {
-      @Override
-      public ICluster apply(@Nullable Void param) {
-        return Cluster.this;
-      }
-    });
+      }, MoreExecutors.directExecutor());
+    return clusterFuture.thenApplyAsync(aVoid -> Cluster.this, MoreExecutors.directExecutor());
```

Comment:
Also btw, I don't get one thing. Either it make at all any sense to call `thenApplyAsync` and then pass same thread executor. Would not it be better or the same to just call `thenApply`?